Model/FeatureSwitchRepository.php

```diff
@@ -26,6 +26,30 @@ public function getByName($name) {
         return $switch;
     }
 
+    /**
+     * Either creates a new switch or updates an existing switch.
+     *
+     * @param string $name
+     * @param boolean $value
+     * @param boolean $defaultValue
+     * @param int $rolloutPercentage
+     * @return \Bolt\Boltpay\Api\Data\FeatureSwitchInterface
+     */
+    public function upsertByName($name, $value, $defaultValue, $rolloutPercentage) {
+        $switch = null;
+        try {
+            $switch = $this->getByName($name);
+        } catch(\Exception $e) {
+            // If the switch is not found we create it.
+            $switch = $this->featureSwitchFactory->create();
+            $switch ->setName($name);
+        }
+        $switch ->setValue($value);
+        $switch ->setDefaultValue($defaultValue);
+        $switch ->setRolloutPercentage($rolloutPercentage);
+        return $this->save($switch);
+    }
+
     public function save(\Bolt\Boltpay\Api\Data\FeatureSwitchInterface $switch) {
         $switch->getResource()->save($switch);
```

Comment:
Does it need to catch some exceptions here?

Reply:
i was considering make it such that the method calling this should catch? lmk what you think